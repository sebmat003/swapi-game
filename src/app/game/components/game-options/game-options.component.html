<mat-form-field>
  <mat-label>Choose property</mat-label>
  <mat-select [formControl]="personProperty">
    <mat-option *ngFor="let topping of ['mass', 'height']" [value]="topping">{{topping}}</mat-option>
  </mat-select>
</mat-form-field>
