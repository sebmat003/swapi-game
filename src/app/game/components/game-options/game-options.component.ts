import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { FormControl } from "@angular/forms";

@Component({
  selector: 'app-game-options',
  templateUrl: './game-options.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class GameOptionsComponent {
  @Input() personProperty!: FormControl;
}
