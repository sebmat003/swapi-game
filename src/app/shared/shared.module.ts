import { NgModule } from '@angular/core';
import { CommonModule } from "@angular/common";
import { LoaderComponent } from "./components/loader/loader.component";
import { IconComponent } from './components/icon/icon.component';
import { ReactiveFormsModule } from "@angular/forms";
import { MatCardModule } from "@angular/material/card";
import { MatSelectModule } from "@angular/material/select";

export const COMPONENTS = [
  LoaderComponent,
  IconComponent
];
export const MODULES = [
  CommonModule,
  ReactiveFormsModule,
  MatCardModule,
  MatSelectModule,
];

@NgModule({
  declarations: [COMPONENTS],
  imports: [MODULES],
  exports: [COMPONENTS, MODULES]
})
export class SharedModule {
}
