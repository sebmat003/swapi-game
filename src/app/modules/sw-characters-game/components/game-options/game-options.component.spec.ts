import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { GameOptionsComponent } from './game-options.component';
import { FormControl } from '@angular/forms';
import { By } from '@angular/platform-browser';
import { SharedModule } from '../../../../shared/shared.module';

describe('GameOptionsComponent', () => {
  let spectator: Spectator<GameOptionsComponent>;
  const createComponent = createComponentFactory({
    component: GameOptionsComponent,
    imports: [SharedModule],
    detectChanges: false,
  });

  beforeEach(() => {
    spectator = createComponent();

    spectator.setInput('personProperty', new FormControl(null));
  });

  it('Should match snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('Should set attribute in ng-select', () => {
    spectator.detectChanges();

    spectator.click(
      spectator.debugElement.query(By.css('.mat-mdc-select-trigger'))
        .nativeElement,
    );
    spectator.detectChanges();
    spectator.click(
      spectator.debugElement.query(By.css('.mat-mdc-option')).nativeElement,
    );

    expect(spectator.component.personProperty.value).toEqual('mass');
  });
});
