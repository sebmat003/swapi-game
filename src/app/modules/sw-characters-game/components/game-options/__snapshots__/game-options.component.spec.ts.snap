// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOptionsComponent Should match snapshot 1`] = `
<app-game-options
  personProperty={[Function FormControl2]}
>
  <mat-form-field
    class="mat-mdc-form-field mat-mdc-form-field-type-mat-select mat-form-field-no-animations mat-form-field-appearance-fill mat-form-field-hide-placeholder mat-primary ng-untouched ng-pristine ng-valid"
  >
    <div
      class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--filled"
    >
      <div
        class="mat-mdc-form-field-focus-overlay"
      />
      <div
        class="mat-mdc-form-field-flex"
      >
        <div
          class="mat-mdc-form-field-infix"
        >
          <label
            class="mdc-floating-label mat-mdc-floating-label"
            matformfieldfloatinglabel=""
          >
            <mat-label>
              Choose property
            </mat-label>
          </label>
          <mat-select
            aria-autocomplete="none"
            aria-disabled="false"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-invalid="false"
            aria-required="false"
            class="mat-mdc-select mat-mdc-select-empty ng-untouched ng-pristine ng-valid"
            data-cy="choose-property-select"
            ngskiphydration=""
            role="combobox"
            tabindex="0"
          >
            <div
              cdk-overlay-origin=""
              class="mat-mdc-select-trigger"
            >
              <div
                class="mat-mdc-select-value"
              >
                <span
                  class="mat-mdc-select-placeholder mat-mdc-select-min-line"
                >
                </span>
              </div>
              <div
                class="mat-mdc-select-arrow-wrapper"
              >
                <div
                  class="mat-mdc-select-arrow"
                >
                  <svg
                    aria-hidden="true"
                    focusable="false"
                    height="24px"
                    viewBox="0 0 24 24"
                    width="24px"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </mat-select>
        </div>
      </div>
      <div
        class="mdc-line-ripple mdc-line-ripple--deactivating"
        matformfieldlineripple=""
      />
    </div>
    <div
      class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
    >
      <div
        class="mat-mdc-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
        style="opacity: 1; transform: translateY(0%);"
      >
        <div
          class="mat-mdc-form-field-hint-spacer"
        />
      </div>
    </div>
  </mat-form-field>
</app-game-options>
`;
